### ISSUE TITLE
Yolov5 auto-update installs packages to local python instead of the current environment I am using

### ISSUE THREAD

**Issue by edehino** [ultralytics/yolov5#6943]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

Hi,

I have a conda environment created with the following packages installed:

Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
_openmp_mutex             4.5                       1_gnu  
_tflow_select             2.1.0                       gpu  
absl-py                   0.15.0             pyhd3eb1b0_0  
aiohttp                   3.8.1            py39h7f8727e_1  
aiosignal                 1.2.0              pyhd3eb1b0_0  
argon2-cffi               21.3.0             pyhd3eb1b0_0  
argon2-cffi-bindings      21.2.0           py39h7f8727e_0  
astor                     0.8.1            py39h06a4308_0  
astunparse                1.6.3                      py_0  
async-timeout             4.0.1              pyhd3eb1b0_0  
attrs                     21.4.0             pyhd3eb1b0_0  
backcall                  0.2.0              pyhd3eb1b0_0  
blas                      1.0                         mkl  
bleach                    4.1.0              pyhd3eb1b0_0  
blinker                   1.4              py39h06a4308_0  
brotlipy                  0.7.0           py39h27cfd23_1003  
c-ares                    1.18.1               h7f8727e_0  
ca-certificates           2022.2.1             h06a4308_0  
cachetools                4.2.2              pyhd3eb1b0_0  
certifi                   2021.10.8        py39h06a4308_2  
cffi                      1.15.0           py39hd667e15_1  
charset-normalizer        2.0.4              pyhd3eb1b0_0  
click                     8.0.4            py39h06a4308_0  
cryptography              3.4.8            py39hd23ed53_0  
cudatoolkit               10.1.243             h6bb024c_0  
cudnn                     7.6.5                cuda10.1_0  
cupti                     10.1.168                      0  
cycler                    0.11.0                   pypi_0    pypi
dataclasses               0.8                pyh6d0b6a4_7  
debugpy                   1.5.1            py39h295c915_0  
decorator                 5.1.1              pyhd3eb1b0_0  
defusedxml                0.7.1              pyhd3eb1b0_0  
docker-pycreds            0.4.0                    pypi_0    pypi
entrypoints               0.3              py39h06a4308_0  
flatbuffers               1.12                     pypi_0    pypi
fonttools                 4.30.0                   pypi_0    pypi
frozenlist                1.2.0            py39h7f8727e_0  
gast                      0.3.3                    pypi_0    pypi
gitdb                     4.0.9                    pypi_0    pypi
gitpython                 3.1.27                   pypi_0    pypi
google-auth               2.6.0              pyhd3eb1b0_0  
google-auth-oauthlib      0.4.4              pyhd3eb1b0_0  
google-pasta              0.2.0              pyhd3eb1b0_0  
grpcio                    1.32.0                   pypi_0    pypi
h5py                      2.10.0           py39hec9cf62_0  
hdf5                      1.10.6               hb1b8bf9_0  
idna                      3.3                pyhd3eb1b0_0  
importlib-metadata        4.8.2            py39h06a4308_0  
importlib_metadata        4.8.2                hd3eb1b0_0  
intel-openmp              2021.4.0          h06a4308_3561  
ipykernel                 6.4.1            py39h06a4308_1  
ipython                   7.31.1           py39h06a4308_0  
ipython_genutils          0.2.0              pyhd3eb1b0_1  
jedi                      0.18.1           py39h06a4308_1  
jinja2                    3.0.2              pyhd3eb1b0_0  
jsonschema                3.2.0              pyhd3eb1b0_2  
jupyter_client            7.1.2              pyhd3eb1b0_0  
jupyter_core              4.9.1            py39h06a4308_0  
jupyterlab_pygments       0.1.2                      py_0  
keras-preprocessing       1.1.2              pyhd3eb1b0_0  
kiwisolver                1.3.2                    pypi_0    pypi
ld_impl_linux-64          2.35.1               h7274673_9  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.3.0               h5101ec6_17  
libgfortran-ng            7.5.0               ha8ba4b0_17  
libgfortran4              7.5.0               ha8ba4b0_17  
libgomp                   9.3.0               h5101ec6_17  
libprotobuf               3.19.1               h4ff587b_0  
libsodium                 1.0.18               h7b6447c_0  
libstdcxx-ng              9.3.0               hd4cf53a_17  
markdown                  3.3.4            py39h06a4308_0  
markupsafe                2.0.1            py39h27cfd23_0  
matplotlib                3.5.1                    pypi_0    pypi
matplotlib-inline         0.1.2              pyhd3eb1b0_2  
mistune                   0.8.4           py39h27cfd23_1000  
mkl                       2021.4.0           h06a4308_640  
mkl-service               2.4.0            py39h7f8727e_0  
mkl_fft                   1.3.1            py39hd3c417c_0  
mkl_random                1.2.2            py39h51133e4_0  
multidict                 5.2.0            py39h7f8727e_2  
nbclient                  0.5.11             pyhd3eb1b0_0  
nbconvert                 6.3.0            py39h06a4308_0  
nbformat                  5.1.3              pyhd3eb1b0_0  
ncurses                   6.3                  h7f8727e_2  
nest-asyncio              1.5.1              pyhd3eb1b0_0  
notebook                  6.4.8            py39h06a4308_0  
numpy                     1.19.5                   pypi_0    pypi
numpy-base                1.21.2           py39h79a1101_0  
oauthlib                  3.2.0              pyhd3eb1b0_0  
opencv-python             4.5.5.64                 pypi_0    pypi
openssl                   1.1.1m               h7f8727e_0  
opt_einsum                3.3.0              pyhd3eb1b0_1  
packaging                 21.3               pyhd3eb1b0_0  
pandas                    1.4.1                    pypi_0    pypi
pandocfilters             1.5.0              pyhd3eb1b0_0  
parso                     0.8.3              pyhd3eb1b0_0  
pathtools                 0.1.2                    pypi_0    pypi
pexpect                   4.8.0              pyhd3eb1b0_3  
pickleshare               0.7.5           pyhd3eb1b0_1003  
pillow                    9.0.1                    pypi_0    pypi
pip                       21.2.4           py39h06a4308_0  
prometheus_client         0.13.1             pyhd3eb1b0_0  
promise                   2.3                      pypi_0    pypi
prompt-toolkit            3.0.20             pyhd3eb1b0_0  
protobuf                  3.19.1           py39h295c915_0  
psutil                    5.9.0                    pypi_0    pypi
ptyprocess                0.7.0              pyhd3eb1b0_2  
pyasn1                    0.4.8              pyhd3eb1b0_0  
pyasn1-modules            0.2.8                      py_0  
pycparser                 2.21               pyhd3eb1b0_0  
pygments                  2.11.2             pyhd3eb1b0_0  
pyjwt                     2.1.0            py39h06a4308_0  
pyopenssl                 21.0.0             pyhd3eb1b0_1  
pyparsing                 3.0.7                    pypi_0    pypi
pyrsistent                0.18.0           py39heee7806_0  
pysocks                   1.7.1            py39h06a4308_0  
python                    3.9.7                h12debd9_1  
python-dateutil           2.8.2              pyhd3eb1b0_0  
python-flatbuffers        2.0                pyhd3eb1b0_0  
pytz                      2021.3                   pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
pyzmq                     22.3.0           py39h295c915_2  
readline                  8.1.2                h7f8727e_1  
requests                  2.27.1             pyhd3eb1b0_0  
requests-oauthlib         1.3.0                      py_0  
rsa                       4.7.2              pyhd3eb1b0_1  
scipy                     1.7.3            py39hc147768_0  
seaborn                   0.11.2                   pypi_0    pypi
send2trash                1.8.0              pyhd3eb1b0_1  
sentry-sdk                1.5.7                    pypi_0    pypi
setproctitle              1.2.2                    pypi_0    pypi
setuptools                58.0.4           py39h06a4308_0  
shortuuid                 1.0.8                    pypi_0    pypi
six                       1.15.0                   pypi_0    pypi
smmap                     5.0.0                    pypi_0    pypi
sqlite                    3.38.0               hc218d9a_0  
**tensorboard               2.8.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.6.0                      py_0  
tensorflow                2.4.1           gpu_py39h8236f22_0  
tensorflow-base           2.4.1**           **gpu_py39h29c2da4_0  
tensorflow-estimator      2.4.0                    pypi_0    pypi
tensorflow-gpu            2.4.1                h30adc30_0**  
termcolor                 1.1.0            py39h06a4308_1  
terminado                 0.13.1           py39h06a4308_0  
testpath                  0.5.0              pyhd3eb1b0_0  
thop                      0.0.31-2005241907          pypi_0    pypi
tk                        8.6.11               h1ccaba5_0  
torch                     1.11.0                   pypi_0    pypi
torchvision               0.12.0                   pypi_0    pypi
tornado                   6.1              py39h27cfd23_0  
tqdm                      4.63.0                   pypi_0    pypi
traitlets                 5.1.1              pyhd3eb1b0_0  
typing-extensions         3.7.4.3                  pypi_0    pypi
typing_extensions         3.10.0.2           pyh06a4308_0  
tzdata                    2021e                hda174b7_0  
urllib3                   1.26.8             pyhd3eb1b0_0  
wandb                     0.12.11                  pypi_0    pypi
wcwidth                   0.2.5              pyhd3eb1b0_0  
webencodings              0.5.1            py39h06a4308_1  
werkzeug                  2.0.3              pyhd3eb1b0_0  
wheel                     0.37.1             pyhd3eb1b0_0  
widgetsnbextension        3.5.2            py39h06a4308_0  
wrapt                     1.12.1                   pypi_0    pypi
xz                        5.2.5                h7b6447c_0  
yarl                      1.6.3            py39h27cfd23_0  
yaspin                    2.1.0                    pypi_0    pypi
zeromq                    4.3.4                h2531618_0  
zipp                      3.7.0              pyhd3eb1b0_0  
zlib                      1.2.11               h7f8727e_4 

When I try to run train.py, there's a prompt that I don't have tensorflow>=2.4.1 but as you can see above, I do have it inside my conda environment.

`requirements: tensorflow>=2.4.1 not found and is required by YOLOv5, attempting auto-update...
tensorflow-gpu 2.1.0 requires gast==0.2.2, but you have gast 0.4.0 which is incompatible.
tensorflow-gpu 2.1.0 requires tensorboard<2.2.0,>=2.1.0, but you have tensorboard 2.6.0 which is incompatible.
tensorflow-gpu 2.1.0 requires tensorflow-estimator<2.2.0,>=2.1.0rc0, but you have tensorflow-estimator 2.6.0 which is incompatible.
Defaulting to user installation because normal site-packages is not writeable
Collecting tensorflow>=2.4.1
  Downloading tensorflow-2.6.2-cp36-cp36m-manylinux2010_x86_64.whl (458.3 MB)
Requirement already satisfied: opt-einsum~=3.3.0 in /home/usr/.local/lib/python3.6/site-packages (from tensorflow>=2.4.1) (3.3.0)`

Take note, I added the conda environment to my jupyter kernel using  `python -m ipykernel install --user --name=env` and I made sure I can see the kernel inside my notebook.

I am not sure if this is a problem on my end but anything that could help me remove this error would be very much appreciated. Thanks!

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@edehino this issue has long ago been resolved. Your YOLOv5 code is out of date. To update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
